### ISSUE TITLE
Gridmap/MeshLib not working with GPUParticlesCollision nodes

### ISSUE THREAD

**Issue by MikeLundahl** [godotengine/godot#86294]
### Tested versions

- Godot Version 4.1.1 and 4.2.1

### System information

Godot v4.1.1.stable - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1070 Ti (NVIDIA; 31.0.15.4584) - Intel(R) Core(TM) i7-5930K CPU @ 3.50GHz (12 Threads)

### Issue description

The gridmap works with normal collision shapes, but not with collision with particles like GPUParticlesCollision nodes. 
![particle_test](https://github.com/godotengine/godot/assets/42893106/534f8917-ce30-4c17-af6d-378523012be4)

### Steps to reproduce

1. create a 3D scene with the gridmap assets and make sure it includes both collision shape and GPUParticlesCollisionBox
2. export the scene as a meshlib
3. create a new 3d scene and create a gridmap.
4. create GPUParticles3D and make sure the particles are in range to collide with the gridmap

### Minimal reproduction project (MRP)

[particle_test.zip](https://github.com/godotengine/godot/files/13702963/particle_test.zip)

**Comment by smix8** [comment]
It is rendering "GPU Collision" from the RenderingServer.

Even if there is collision in this name it has nothing to do with physics collision of the PhysicsServer.

**Comment by Calinou** [comment]
This would require adding support for instantiating GPUParticlesCollision* nodes from a GridMap, which could get pretty slow. I suggest baking a single GPUParticlesCollisionbSDF3D after designing your GridMap instead, or using GPUParticlesCollisionHeightField3D.

**Comment by MikeLundahl** [comment]
GPUParticlesCollisionSDF3D could be a good solution. However (reading the documents), it only supports baking from the editor. So if someone, like in my case, has created a procedural level system using gridmaps (through GDSCript) that generates the gridmap during runtime. The GPUParticlesCollisionbSDF3D would also need to be baked through scripting.

**Comment by NuclearPhoenixx** [comment]
Yes, it looks like GPUParticlesCollisionSDF3D need baking of a static scene. The downside of GPUParticlesCollisionHeightField3D is obviously that it doesn't work with any kind of overhang.

Maybe it would be possible to grab the surface area of the GridMap and dynamically place GPUParticlesCollision nodes inside of them to emulate the export? I'm guessing performance wouldn't be too great though and it would add significant time to pre-compute all of that when loading the geometry for the first time.

Off topic, but the exact same thing is true for the OccluderInstance3D. Cannot bake it for dynamic scenes that _could_ change and it's not included in the MeshLibrary on export.
